src/layout/elkLayout.ts

```diff
@@ -5,24 +5,163 @@ import { formatText, measureText } from './utils';
 
 export type CanvasDirection = 'LEFT' | 'RIGHT' | 'DOWN' | 'UP';
 
-export interface ElkOptions {
-  direction: CanvasDirection;
+/**
+ * ELKjs layout options for the Canvas.
+ *
+ * Unfortunately, the ELKjs documentation is not straightforward.
+ * You'll likely need to take a look at the ELK options reference to see all available options.
+ *
+ * @see https://github.com/kieler/elkjs#layout-options
+ * @see https://www.eclipse.org/elk/reference/options.html
+ */
+export interface ElkCanvasLayoutOptions {
+  'elk.direction'?: CanvasDirection;
+  [key: string]: string;
 }
 
-const defaultLayoutOptions = {
+/**
+ * ELKjs layout option for a node.
+ *
+ * TODO add reference link, I don't know what are the available options.
+ */
+export interface ElkNodeLayoutOptions {
+  [key: string]: string;
+}
+
+/**
+ * ELK layout options applied by default, unless overridden through <Canvas layoutOptions> property.
+ *
+ * XXX Not to be confounded with ELK "defaultLayoutOptions" property, which is meant to be used as fallback, when no layout option is provided.
+ *
+ * @see https://www.eclipse.org/elk/reference/options.html
+ */
+const defaultLayoutOptions: ElkCanvasLayoutOptions = {
+  /**
+   * Hints for where node labels are to be placed; if empty, the node label’s position is not modified.
+   *
+   * @see https://www.eclipse.org/elk/reference/options/org-eclipse-elk-nodeLabels-placement.html
+   */
   'elk.nodeLabels.placement': 'INSIDE V_CENTER H_RIGHT',
+
+  /**
+   * Select a specific layout algorithm.
+   *
+   * Uses "layered" strategy.
+   * It emphasizes the direction of edges by pointing as many edges as possible into the same direction.
+   * The nodes are arranged in layers, which are sometimes called “hierarchies”,
+   * and then reordered such that the number of edge crossings is minimized.
+   * Afterwards, concrete coordinates are computed for the nodes and edge bend points.
+   *
+   * @see https://www.eclipse.org/elk/reference/algorithms.html
+   * @see https://www.eclipse.org/elk/reference/options/org-eclipse-elk-algorithm.html
+   * @see https://www.eclipse.org/elk/reference/algorithms/org-eclipse-elk-layered.html
+   */
   'elk.algorithm': 'org.eclipse.elk.layered',
+
+  /**
+   * Overall direction of edges: horizontal (right / left) or vertical (down / up).
+   *
+   * @see https://www.eclipse.org/elk/reference/options/org-eclipse-elk-direction.html
+   */
   'elk.direction': 'DOWN',
+
+  /**
+   * XXX NOT SURE - Option doesn't seem to exist, should be "org.eclipse.elk.layered.layering.strategy"?
+   * Strategy for node layering.
+   *
+   * @see https://www.eclipse.org/elk/reference/options/org-eclipse-elk-layered-layering-strategy.html
+   */
   nodeLayering: 'INTERACTIVE',
```

Comment:
@amcdnl This option seems wrong and will cause a breaking change when updating elkjs.

See https://github.com/kieler/elkjs/issues/122#issuecomment-778803355

It should be renamed to `org.eclipse.elk.layered.layering.strategy`, if I understand it well.

Reply:
Good w/ the change